### PR TITLE
Fix pygrib openfiles

### PR DESCRIPTION
**Pull request by adybbroe**
- [ ] Closes #xxxx 
 - [ ] Tests added 
 - [ ] Tests passed: Passes ``pytest pyspectral`` 
 - [ ] Passes ``flake8 pyspectral`` 
 - [ ] Fully documented

### CODE REVIEW

File: nwcsafpps_runner/prepare_nwp.py [hunk 1]
```diff
@@ -157,8 +155,10 @@ def update_nwp(starttime, nlengths):
             LOG.info("File: " + str(result_file) + " already there...")
             continue
 
-        dummy, tmp_filename = tempfile.mkstemp(suffix="_" + timestamp + "+" +
-                                               '%.3dH00M' % forecast_step, dir=nwp_outdir)
+        tmp_filename_handle, tmp_filename = tempfile.mkstemp(suffix="_" + timestamp + "+" +
+                                                             '%.3dH00M' % forecast_step, dir=nwp_outdir)
+        os.close(tmp_filename_handle)
+
```
Comment (mraspaud) [inline nwcsafpps_runner/prepare_nwp.py:161]: you could make a small function instead to return a temporary filename, eg:
```python
def make_temp_filename(*args, **kwargs):
    tmp_filename_handle, tmp_filename = tempfile.mkstemp(*args, **kwargs)
    os.close(tmp_filename_handle)
    return tmp_filename
```

File: nwcsafpps_runner/prepare_nwp.py [hunk 2]
```diff
@@ -182,34 +182,46 @@ def update_nwp(starttime, nlengths):
                       "topography available. Can't prepare NWP data")
             raise IOError('Failed getting static land-sea mask and topography')
 
-        dummy, tmp_result_filename = tempfile.mkstemp()
+        tmp_result_filename_handle, tmp_result_filename = tempfile.mkstemp()
+        os.close(tmp_result_filename_handle)
```
Comment (mraspaud) [inline nwcsafpps_runner/prepare_nwp.py:186]: And use the new function here too.

**Review by mraspaud — COMMENTED** [review]
Looks like this code some TLC, but this is good enough for now I guess.
Do we really need all these new debug messages?
Also, let's make a function for getting a temporary filename.

File: nwcsafpps_runner/prepare_nwp.py (hunk 1 above)
Comment (adybbroe) [reply nwcsafpps_runner/prepare_nwp.py:161]: Yes, good idea, I will do

Comment (adybbroe) [reply nwcsafpps_runner/prepare_nwp.py:161]: TLC?

Comment (mraspaud) [reply nwcsafpps_runner/prepare_nwp.py:161]: Tender Love and Care

Comment (adybbroe) [reply nwcsafpps_runner/prepare_nwp.py:161]: Function added and used.
Concerning debug-messages, you are right. I have removed a few.

File: nwcsafpps_runner/prepare_nwp.py (hunk 2 above)
Comment (adybbroe) [reply nwcsafpps_runner/prepare_nwp.py:186]: Done

**Comment by mraspaud** [comment]
LGTM, feel free to merge once the tests pass.

**Comment by adybbroe** [comment]
Thanks @mraspaud 
I think it will fail on Ubuntu-latest. Something with gdal-config. I couldn't really see what I could do to make it pass, so I left it for now. Already a good step forward to have testing on github I thought.

**Comment by adybbroe** [comment]
Oh, it also fails on Windows, as I have hardcoded Linux style filepaths in the tests. The code had never run under Windows yet, as far as we know, so I will ignore this here, and address it in a separate PR

**Pull request merged**